src/components/header/header.tsx

```diff
@@ -135,6 +135,12 @@ const Header: React.FC<HeaderProps> = (props) => {
   );
   const [hoverWithTransition, setHoverWithTransition] = useState(true);
 
+  const transparent = Boolean(props.transparent);
+
+  useEffect(() => {
```

Comment:
thank you! Have you checked the docs to understand why this is working?
Here the excellent docs: https://reactjs.org/docs/hooks-effect.html and the comparison with `componentDidMount` & friends from the classic react world